packages/components/src/hooks/use-debounce.ts

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import debounce from 'lodash.debounce';
+
+export const useDebounce = (callback: () => void, timeout: number) => {
+    const debouncedCallback = React.useMemo(
+        () =>
+            debounce(() => {
+                callback();
+            }, timeout),
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        []
+    );
+
+    React.useEffect(() => {
+        return debouncedCallback.cancel;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
what does this do? May be it's not needed :D

Reply:
Okay. I understand the logic now.

Reply:
Can we write something like the below without using the debounce library from lodash?
The below one is easier to understand and maybe we can remove the lodash debounce in the future.

```
import { useEffect, useRef } from 'react';

const useDebounce = (callback, delay) => {
  const timeoutRef = useRef(null);

  useEffect(() => {
    // Cleanup the previous timeout on re-render
    return () => {
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
    };
  }, []);

  const debouncedCallback = (...args) => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current);
    }

    timeoutRef.current = setTimeout(() => {
      callback(...args);
    }, delay);
  };

  return debouncedCallback;
};
```

Reply:
Yes, we can use `setTimeout`, but debounce seems so to be convenient tool. No need in refs and you can clean just by calling `.cancel()`
Also, we can make it shorter:
```
export const useDebounce = (callback: () => void, timeout: number) => {
    const debouncedCallback = React.useMemo(
        () => debounce(callback, timeout),
        // eslint-disable-next-line react-hooks/exhaustive-deps
        []
    );

    React.useEffect(() => {
        return debouncedCallback.cancel;
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return debouncedCallback;
};
```
Should I apply `setTimeout`?

Reply:
I see. Let's go with this version without refs and timeout. Thanks!